packages/profiles/test/musig/btc.musig.registration.test.ts

```diff
@@ -0,0 +1,292 @@
+import { IProfile, IReadWriteWallet } from "../../source/contracts";
+import nock from "nock";
+
+import { mockMusigServer } from "./musig.test.helpers";
+import { mnemonics as testMnemonics } from "../fixtures/identity";
+import { Signatories } from "@payvo/sdk";
+import { oneSignatureNativeSegwitMusigRegistrationTx } from "../../../btc/test/fixtures/musig-native-segwit-txs";
+
+jest.setTimeout(60_000);
+
+describe("BTC", () => {
+	let profile1, profile2, profile3: IProfile;
+
+	const { mockServerResponse, resetServerResponseMocks } = mockMusigServer({
+		url: "https://btc-test-musig.payvo.com",
+	});
+
+	beforeAll(() => {
+		bootContainer();
+
+		// Default mocks
+		nock("https://btc-test.payvo.com:443")
+			// .get("/api/blockchain")
+			// .reply(200, require("../fixtures/client/blockchain.json"))
+			// .get("/api/node/configuration")
+			// .reply(200, require("../fixtures/client/configuration.json"))
+			// .get("/api/peers")
+			// .reply(200, require("../fixtures/client/peers.json"))
+			// .get("/api/node/configuration/crypto")
+			// .reply(200, require("../fixtures/client/cryptoConfiguration.json"))
+			// .get("/api/node/syncing")
+			// .reply(200, require("../fixtures/client/syncing.json"))
+			// // Wallet mocks
+			// .get("/api/wallets/DABCrsfEqhtdzmBrE2AU5NNmdUFCGXKEkr")
+			// .reply(200, require("../fixtures/wallets/DABCrsfEqhtdzmBrE2AU5NNmdUFCGXKEkr.json"))
+			// .get("/api/wallets/DCX2kvwgL2mrd9GjyYAbfXLGGXWwgN3Px7")
+			// .reply(200, require("../fixtures/wallets/DCX2kvwgL2mrd9GjyYAbfXLGGXWwgN3Px7.json"))
+			// .get("/api/wallets/DDHk393YcsxTPN1H5SWTcbjfnCRmF1iBR8")
+			// .reply(200, require("../fixtures/wallets/DDHk393YcsxTPN1H5SWTcbjfnCRmF1iBR8.json"))
+			.persist();
+
+		profile1 = makeProfile({ id: "profile-id-1" });
+		profile2 = makeProfile({ id: "profile-id-2" });
+		profile3 = makeProfile({ id: "profile-id-3" });
+	});
+
+	afterEach(() => {
+		resetServerResponseMocks();
+	});
+
+	describe("MuSig Registration", () => {
+		it("should perform a 2 out of 2 registration", async () => {
+			// TODO We shouldn't need to create a wallet first, but DW requires so, as the button to create musig wallet is inside an existing wallet
+			const first = await bitcoinWallet({
+				profile: profile1,
+				coinId: "BTC",
+				networkId: "btc.testnet",
+				mnemonic: testMnemonics["btc.testnet"][0],
+				account: 0,
+				// path: "m/48'/1'/0'/2'",
+			});
+
+			const signatory1 = new Signatories.Signatory(
+				new Signatories.MnemonicSignatory({
+					signingKey: testMnemonics["btc.testnet"][0],
+					address: "address", // Not needed / used
+					publicKey: "m/48'/1'/0'/2'", // TODO for now we use publicKey for passing path
+					privateKey: "privateKey", // Not needed / used
+				}),
+			);
+
+			const transactionData = {
+				signatory: signatory1,
+				data: {
+					min: 2,
+					numberOfSignatures: 2,
+					publicKeys: [],
+					derivationMethod: "nativeSegwitMusig",
```

Comment:
@marianogoldman creating an enum as such 
```ts
declare enum MusigDerivationMethod {
	LegacyMuSig = "legacyMusig",
	P2SHSegwitMusig = "p2SHSegwitMusig",
	NativeSegwitMusig = "nativeSegwitMusig",
}

```
add add it as `derivationMethod` instead of string should fix the type error in `signMultiSignature`

Reply:
👍